openeo/rest/connection.py

```diff
@@ -55,6 +55,34 @@ def __init__(self, http_status_code: int = None,
         self.url = url
         super().__init__("[{s}] {c}: {m}".format(s=self.http_status_code, c=self.code, m=self.message))
 
+class Paginate:
```

Comment:
I think this can be implemented more succinctly  with a generator (`yield`) instead of implementing the iterator protocol.

Something like this (untested):

```python

def paginate(con: "Connection", url: str, callback: Callback):
    page = 0
    while True:
        resp = con.get(url).json()
        yield callback(resp, page)
        next_links = [link for link in resp.get("links", []) if link["rel"] == "next] 
        if not next_links:
             break
        url = next_links[0]
        page += 1
        


```

Reply:
Does that return? I guess I need to learn what yield is before...

Reply:
How would the method be used then? Is it the same as before?

Reply:
yes both are basically equivalent,
 `yield` allows you to create an (lazy) iterator in a more to-the-point way than the iterator protocol (`__iter__` and `_next__`)

Reply:
Implemented, works, thanks.